aiida_firecrest/transport.py

```diff
@@ -315,45 +564,75 @@ def copy(
+        # TODO: investigate overwrite (?)
+
         if not recursive:
             # TODO this appears to not actually be used upstream, so just remove there
             raise NotImplementedError("Non-recursive copy not implemented")
-        source = self._cwd.joinpath(remotesource).enable_cache()
-        destination = self._cwd.joinpath(remotedestination).enable_cache()
+        if dereference:
+            raise NotImplementedError(
+                "Dereferencing not implemented in FirecREST server"
+            )
+        source = self._cwd.joinpath(
+            remotesource
+        )  # .enable_cache() it's removed from from path.py to be investigated
+        destination = self._cwd.joinpath(
+            remotedestination
+        )  # .enable_cache() it's removed from from path.py to be investigated
 
         if not source.exists():
-            raise FileNotFoundError(f"Source file does not exist: {source}")
-
-        if not dereference and source.is_symlink():
-            destination.symlink_to(source)
-        else:
-            source.copy_to(destination)
-
-    def makedirs(self, path: str, ignore_existing: bool = False) -> None:
-        self._cwd.joinpath(path).mkdir(parents=True, exist_ok=ignore_existing)
+            raise FileNotFoundError(f"Source does not exist: {source}")
+        if not destination.exists() and not source.is_file():
+            raise FileNotFoundError(f"Destination does not exist: {destination}")
 
-    def mkdir(self, path: str, ignore_existing: bool = False) -> None:
-        self._cwd.joinpath(path).mkdir(exist_ok=ignore_existing)
+        self._copy_to(source, destination)
 
-    # TODO check symlink handling for get methods
     # TODO do get/put methods need to handle glob patterns?
+    # Apparently not, but I'm not clear how glob() iglob() are going to behave here. We may need to implement them.
 
     def getfile(
-        self, remotepath: str | FcPath, localpath: str | Path, *args: Any, **kwargs: Any
+        self,
+        remotepath: str | FcPath,
+        localpath: str | Path,
+        dereference: bool = True,
+        *args: Any,
+        **kwargs: Any,
     ) -> None:
+        """Get a file from the remote.
+
+        :param dereference: If True, follow symlinks.
+            note: we don't support downloading symlinks, so dereference should always be True
+
+        """
+        if not dereference:
+            raise NotImplementedError(
+                "Getting symlinks with `dereference=False` is not supported"
+            )
+
         local = Path(localpath)
         if not local.is_absolute():
             raise ValueError("Destination must be an absolute path")
         remote = (
             remotepath
             if isinstance(remotepath, FcPath)
-            else self._cwd.joinpath(remotepath).enable_cache()
+            else self._cwd.joinpath(
+                remotepath
+            )  # .enable_cache() it's removed from from path.py to be investigated
```

Comment:
unclear, please consider to make an issue

Reply:
Alright, just opened an issue is opened on this:
https://github.com/aiidateam/aiida-firecrest/issues/41